alpha/apps/kaltura/lib/kSimuliveUtils.php

```diff
@@ -38,9 +38,14 @@ public static function getSimuliveEventDetails(LiveEntry $entry, $time)
 		$startTime = $currentEvent->getCalculatedStartTime() * self::SECOND_IN_MILLISECONDS;
 		$durations[] = min($sourceEntry->getLengthInMsecs(), ($currentEvent->getCalculatedEndTime() * self::SECOND_IN_MILLISECONDS) - $startTime);
 		$endTime = $startTime + array_sum($durations);
+		$assets = array();
 		// getting the flavors from source entry
-		$flavors = assetPeer::retrieveReadyWebByEntryId($sourceEntry->getId());
-		return array($durations, $flavors, $startTime, $endTime, $dvrWindowMs);
+		$flavorAssets = assetPeer::retrieveReadyWebByEntryId($sourceEntry->getId());
+		$assets = array_merge($assets, $flavorAssets);
```

Comment:
Why is this needed? asset is empty.
Why not just create asset out of the merge of flavour and captions?